src/components/Header/Header.tsx

```diff
@@ -0,0 +1,78 @@
+import {
+  FC,
+  useState,
+  useCallback,
+  Dispatch,
+  SetStateAction,
+} from 'react';
+import { createTodo } from '../../api/todos';
+import { Todo } from '../../types/Todo';
+
+type Props = {
+  onAddTodo: Dispatch<SetStateAction<Todo[]>>;
+  setError: (error: boolean) => void;
+};
+
+export const Header: FC<Props> = ({ onAddTodo, setError }) => {
+  const [title, setTitle] = useState('');
+  /* eslint-disable @typescript-eslint/no-non-null-assertion */
+  const user = JSON.parse(localStorage.getItem('user')!) || {
```

Comment:
`!` as `any` in TS better to avoid because it's remove sense of checking.

In your case you can make sth like that:
```
const storedUser = localStorage.getItem('user');
const user = storedUser ? JSON.parse(storedUser) : initialUser;
```

And question, why you don't use `useLocalStorage` here? Looks like it can be helpful here